### ISSUE TITLE
Container running with non vpn ip.

### ISSUE THREAD

**Issue by Freethefire** [DyonR/docker-qbittorrentvpn#111]
Hey, 

Just recreated the container under Portainer and then checked torrent ip, to my horror it showed my personal ip. Not the vpn ip.
I have added a ovpn file to correct folder. Been running this for a few years without issues before.
This is my compose:

```---
version: "2.1"
services:
  qbittorrentvpn:
    image: dyonr/qbittorrentvpn
    container_name: qbittorrentvpn
    privileged: true
    environment:
      - VPN_ENABLED=true
      - VPN_TYPE=openvpn
      - LAN_NETWORK=10.0.0.0/24
      - VPN_USERNAME=XXXXXXXX
      - VPN_PASSWORD=XXXXXXXX
      - UMASK=002
      - HEALTH_CHECK_INTERVAL=300
      - HEALTH_CHECK_HOST=one.one.one.one
      - INSTALL_PYTHON3=yes
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Stockholm
      - AUTO_UPDATE=true #optional
    volumes:
      - /home/media/5tb:/data
      - /home/media/docker/dyonr:/config
    ports:
      - 8080:8080
    restart: unless-stopped
```
How it this even possible?

**Comment by DyonR** [comment]
Can you maybe post the log of the container itself instead of the compose?

**Comment by Freethefire** [comment]
> Can you maybe post the log of the container itself instead of the compose?

Ofc sorry.

```
2022-08-21 13:09:54.264340 [INFO] VPN_ENABLED defined as 'true'
2022-08-21 13:09:54.293157 [INFO] LEGACY_IPTABLES is set to ''
2022-08-21 13:09:54.322769 [INFO] Not making any changes to iptables version
2022-08-21 13:09:54.355002 [INFO] The container is currently running iptables v1.8.7 (nf_tables).
2022-08-21 13:09:54.384144 [INFO] VPN_TYPE defined as 'openvpn'
2022-08-21 13:09:54.423512 [INFO] OpenVPN config file is found at /config/openvpn/no-osl.prod.surfshark.comsurfshark_openvpn_udp.ovpn
dos2unix: converting file /config/openvpn/no-osl.prod.surfshark.comsurfshark_openvpn_udp.ovpn to Unix format...
2022-08-21 13:09:54.459299 [INFO] VPN remote line defined as 'no-osl.prod.surfshark.com 1194'
2022-08-21 13:09:54.488643 [INFO] VPN_REMOTE defined as 'no-osl.prod.surfshark.com'
2022-08-21 13:09:54.517146 [INFO] VPN_PORT defined as '1194'
2022-08-21 13:09:54.545942 [INFO] VPN_PROTOCOL defined as 'udp'
2022-08-21 13:09:54.575552 [INFO] VPN_DEVICE_TYPE defined as 'tun0'
2022-08-21 13:09:54.603932 [INFO] LAN_NETWORK defined as '10.0.0.0/24'
2022-08-21 13:09:54.632614 [WARNING] NAME_SERVERS not defined (via -e NAME_SERVERS), defaulting to CloudFlare and Google name servers
2022-08-21 13:09:54.662143 [INFO] VPN_OPTIONS not defined (via -e VPN_OPTIONS)
2022-08-21 13:09:54.691176 [INFO] Adding 1.1.1.1 to resolv.conf
2022-08-21 13:09:54.720196 [INFO] Adding 8.8.8.8 to resolv.conf
2022-08-21 13:09:54.748520 [INFO] Adding 1.0.0.1 to resolv.conf
2022-08-21 13:09:54.778642 [INFO] Adding 8.8.4.4 to resolv.conf
2022-08-21 13:09:54.814573 [INFO] Python3 not yet installed, installing...

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

12 packages can be upgraded. Run 'apt list --upgradable' to see them.

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  libexpat1 libmpdec3 libpython3-stdlib libpython3.9-minimal
  libpython3.9-stdlib libreadline8 libsqlite3-0 media-types python3-minimal
  python3.9 python3.9-minimal readline-common
Suggested packages:
  python3-doc python3-tk python3-venv python3.9-venv python3.9-doc binutils
  binfmt-support readline-doc
The following NEW packages will be installed:
  libexpat1 libmpdec3 libpython3-stdlib libpython3.9-minimal
  libpython3.9-stdlib libreadline8 libsqlite3-0 media-types python3
  python3-minimal python3.9 python3.9-minimal readline-common
0 upgraded, 13 newly installed, 0 to remove and 12 not upgraded.
Need to get 6259 kB of archives.
After this operation, 22.5 MB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bullseye/main amd64 libpython3.9-minimal amd64 3.9.2-1 [801 kB]
Get:2 http://deb.debian.org/debian bullseye/main amd64 libexpat1 amd64 2.2.10-2+deb11u3 [98.5 kB]
Get:3 http://deb.debian.org/debian bullseye/main amd64 python3.9-minimal amd64 3.9.2-1 [1955 kB]
Get:4 http://deb.debian.org/debian bullseye/main amd64 python3-minimal amd64 3.9.2-3 [38.2 kB]
Get:5 http://deb.debian.org/debian bullseye/main amd64 media-types all 4.0.0 [30.3 kB]
Get:6 http://deb.debian.org/debian bullseye/main amd64 libmpdec3 amd64 2.5.1-1 [87.7 kB]
Get:7 http://deb.debian.org/debian bullseye/main amd64 readline-common all 8.1-1 [73.7 kB]
Get:8 http://deb.debian.org/debian bullseye/main amd64 libreadline8 amd64 8.1-1 [169 kB]
Get:9 http://deb.debian.org/debian bullseye/main amd64 libsqlite3-0 amd64 3.34.1-3 [797 kB]
Get:10 http://deb.debian.org/debian bullseye/main amd64 libpython3.9-stdlib amd64 3.9.2-1 [1684 kB]
Get:11 http://deb.debian.org/debian bullseye/main amd64 python3.9 amd64 3.9.2-1 [466 kB]
Get:12 http://deb.debian.org/debian bullseye/main amd64 libpython3-stdlib amd64 3.9.2-3 [21.4 kB]
Get:13 http://deb.debian.org/debian bullseye/main amd64 python3 amd64 3.9.2-3 [37.9 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 6259 kB in 1s (8640 kB/s)
Selecting previously unselected package libpython3.9-minimal:amd64.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 10522 files and directories currently installed.)
Preparing to unpack .../libpython3.9-minimal_3.9.2-1_amd64.deb ...
Unpacking libpython3.9-minimal:amd64 (3.9.2-1) ...
Selecting previously unselected package libexpat1:amd64.
Preparing to unpack .../libexpat1_2.2.10-2+deb11u3_amd64.deb ...
Unpacking libexpat1:amd64 (2.2.10-2+deb11u3) ...
Selecting previously unselected package python3.9-minimal.
Preparing to unpack .../python3.9-minimal_3.9.2-1_amd64.deb ...
Unpacking python3.9-minimal (3.9.2-1) ...
Setting up libpython3.9-minimal:amd64 (3.9.2-1) ...
Setting up libexpat1:amd64 (2.2.10-2+deb11u3) ...
Setting up python3.9-minimal (3.9.2-1) ...
Selecting previously unselected package python3-minimal.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 10816 files and directories currently installed.)
Preparing to unpack .../0-python3-minimal_3.9.2-3_amd64.deb ...
Unpacking python3-minimal (3.9.2-3) ...
Selecting previously unselected package media-types.
Preparing to unpack .../1-media-types_4.0.0_all.deb ...
Unpacking media-types (4.0.0) ...
Selecting previously unselected package libmpdec3:amd64.
Preparing to unpack .../2-libmpdec3_2.5.1-1_amd64.deb ...
Unpacking libmpdec3:amd64 (2.5.1-1) ...
Selecting previously unselected package readline-common.
Preparing to unpack .../3-readline-common_8.1-1_all.deb ...
Unpacking readline-common (8.1-1) ...
Selecting previously unselected package libreadline8:amd64.
Preparing to unpack .../4-libreadline8_8.1-1_amd64.deb ...
Unpacking libreadline8:amd64 (8.1-1) ...
Selecting previously unselected package libsqlite3-0:amd64.
Preparing to unpack .../5-libsqlite3-0_3.34.1-3_amd64.deb ...
Unpacking libsqlite3-0:amd64 (3.34.1-3) ...
Selecting previously unselected package libpython3.9-stdlib:amd64.
Preparing to unpack .../6-libpython3.9-stdlib_3.9.2-1_amd64.deb ...
Unpacking libpython3.9-stdlib:amd64 (3.9.2-1) ...
Selecting previously unselected package python3.9.
Preparing to unpack .../7-python3.9_3.9.2-1_amd64.deb ...
Unpacking python3.9 (3.9.2-1) ...
Selecting previously unselected package libpython3-stdlib:amd64.
Preparing to unpack .../8-libpython3-stdlib_3.9.2-3_amd64.deb ...
Unpacking libpython3-stdlib:amd64 (3.9.2-3) ...
Setting up python3-minimal (3.9.2-3) ...
Selecting previously unselected package python3.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 11245 files and directories currently installed.)
Preparing to unpack .../python3_3.9.2-3_amd64.deb ...
Unpacking python3 (3.9.2-3) ...
Setting up media-types (4.0.0) ...
Setting up libsqlite3-0:amd64 (3.34.1-3) ...
Setting up libmpdec3:amd64 (2.5.1-1) ...
Setting up readline-common (8.1-1) ...
Setting up libreadline8:amd64 (8.1-1) ...
Setting up libpython3.9-stdlib:amd64 (3.9.2-1) ...
Setting up libpython3-stdlib:amd64 (3.9.2-3) ...
Setting up python3.9 (3.9.2-1) ...
Setting up python3 (3.9.2-3) ...
running python rtupdate hooks for python3.9...
running python post-rtupdate hooks for python3.9...
Processing triggers for libc-bin (2.31-13+deb11u3) ...

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
0 upgraded, 0 newly installed, 0 to remove and 12 not upgraded.
2022-08-21 13:10:06.699259 [WARNING] ENABLE_SSL is set to '', SSL is not enabled. This could cause issues with logging if other apps use the same Cookie name (SID).
2022-08-21 13:10:06.734566 [WARNING] Removing the SSL configuration from the config file...
2022-08-21 13:10:06.767532 [INFO] A group with PGID 1000 does not exist, adding a group called 'qbittorrent' with PGID 1000
2022-08-21 13:10:06.818106 [INFO] An user with PUID 1000 does not exist, adding an user called 'qbittorrent user' with PUID 1000
2022-08-21 13:10:06.871806 [INFO] UMASK defined as '002'
2022-08-21 13:10:06.901223 [INFO] Starting qBittorrent daemon...
Logging to /config/qBittorrent/data/logs/qbittorrent.log.
2022-08-21 13:10:07.939691 [INFO] Started qBittorrent daemon successfully...
2022-08-21 13:10:07.969383 [INFO] qBittorrent PID: 664
2022-08-21 13:10:07.998111 [INFO] HEALTH_CHECK_SILENT is not set. Because this variable is not set, it will be supressed by default
2022-08-21 13:10:08.025940 [WARNING] RESTART_CONTAINER not defined,(via -e RESTART_CONTAINER), defaulting to 'yes'
2022-08-21 13:16:48.480361 [INFO] Received SIGTERM, stopping...
```

**Comment by DyonR** [comment]
Alright, thank you! I'll investigate

**Comment by DyonR** [comment]
Set VPN_ENABLED to "**yes**" and not to "true".  
At the moment it only allows **yes** as a valid input. I'll make changes to the script to allow true/yes/1 as valid inputs and add a warning if the VPN_ENABLED is not set to either of those values.

**Comment by Freethefire** [comment]
> Set VPN_ENABLED to "**yes**" and not to "true". At the moment it only allows **yes** as a valid input. I'll make changes to the script to allow true/yes/1 as valid inputs and add a warning if the VPN_ENABLED is not set to either of those values.

Thanks!
But I'm more worried about that the container got internet access even without using the VPN. Shouldn't that be impossible because of the "kill switch"?

**Comment by DyonR** [comment]
The 'kill-switch' only gets enabled if the VPN_ENABLED is properly triggered

**Comment by Freethefire** [comment]
> The 'kill-switch' only gets enabled if the VPN_ENABLED is properly triggered

Oh ok, thanks again for such quick response 👍

**Comment by DyonR** [comment]
I totally forgot that a fix for this was already fixed with #109
I'll merge dev into master now, so in the future it won't cause trouble anymore
